icu4c/source/i18n/unum.cpp

```diff
@@ -701,8 +704,7 @@ unum_getTextAttribute(const UNumberFormat*  fmt,
             *status = U_UNSUPPORTED_ERROR;
             return -1;
         }
-    } else {
-        const RuleBasedNumberFormat* rbnf = dynamic_cast<const RuleBasedNumberFormat*>(nf);
+    } else  if (rbnf != nullptr) {
```

Comment:
nit: consider doing
```suggestion
    } else  if ((rbnf = dynamic_cast<const RuleBasedNumberFormat*>(nf)) != nullptr) {
```
so that we need not spend the small amount of dynamic_cast time in the common case of it being a simple DecimalFormat

Reply:
Not crazy about this, but I'll do it...